src/native/managed/cdacreader/src/Entrypoints.cs

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+
+namespace Microsoft.Diagnostics.DataContractReader;
+
+internal static class Entrypoints
+{
+    private const string CDAC = "cdac_reader_";
+
+    [UnmanagedCallersOnly(EntryPoint = $"{CDAC}init")]
+    private static unsafe int Init(nint descriptor, IntPtr* handle)
+    {
+        Target target = new(descriptor);
+        GCHandle gcHandle = GCHandle.Alloc(target);
+        *handle = GCHandle.ToIntPtr(gcHandle);
+        return 0;
+    }
+
+    [UnmanagedCallersOnly(EntryPoint = $"{CDAC}free")]
+    private static unsafe int Free(IntPtr handle)
+    {
+        GCHandle h = GCHandle.FromIntPtr(handle);
+        h.Free();
+        return 0;
+    }
+
+    /// <summary>
+    /// Get the SOS-DAC interface implementation.
+    /// </summary>
+    /// <param name="handle">Handle crated via cdac initialization</param>
+    /// <param name="obj"><c>IUnknown</c> pointer that can be queried for ISOSDacInterface*</param>
+    /// <returns></returns>
+    [UnmanagedCallersOnly(EntryPoint = $"{CDAC}get_sos_interface")]
+    private static unsafe int GetSOSInterface(IntPtr handle, nint* obj)
+    {
+        ComWrappers cw = new StrategyBasedComWrappers();
+        Target? target = GCHandle.FromIntPtr(handle).Target as Target;
+        if (target == null)
+            return -1;
```

Comment:
Do we want to adopt an `HRESULT` pattern for error codes or more of a POSIX pattern?

Reply:
Is the `Target` class the data target (like `ICLRDataTarget`) that will have read memory etc. methods on it?

Reply:
It's `Target` from https://github.com/dotnet/runtime/blob/main/docs/design/datacontracts/contract_csharp_api_design.cs#L85

it's the cDAC's internal abstraction for the target process

Reply:
How does hosting code provide a ICLRDataTarget instance?

Reply:
That's not wired up yet in this PR, but in the previous prototypes the cDAC exported a function like:

```c
typedef cdac_reader_result_t (*cdac_reader_func_t) (cdac_reader_foreignptr_t addr, uint32_t count, void* user_data, uint8_t *dest);

cdac_reader_result_t cdac_reader_set_reader_func(cdac_reader_h handle, cdac_reader_func_t reader, void* user_data);
```

that was called from the existing DAC, passing a callback that wrapped `ReadDataFromTarget`:

https://github.com/lambdageek/runtime/blob/ab6e9603222842034c4f4452816e14371db450d9/src/coreclr/debug/daccess/cdac.cpp#L262-L296